### ISSUE TITLE
Clang suggests wrong attribute placement for enum classes

### ISSUE THREAD

**Issue by tbaederr** [llvm/llvm-project#61660]
Consider this code:

```c++
__attribute__((deprecated))
enum  class F {
  Foo,
  Bar
};
```

clang suggests:
```console
<source>:3:16: warning: attribute 'deprecated' is ignored, place it after "enum" to apply attribute to type declaration [-Wignored-attributes]
__attribute__((deprecated))
```
Godbolt: https://godbolt.org/z/E4vT1dc57

But placing them after "enum" is wrong, too:

```console
<source>:4:6: error: an attribute list cannot appear here
enum __attribute__((deprecated)) class F {
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Godbolt: https://godbolt.org/z/PWvd8jsr5

**Comment by sudip-unb** [comment]
Interested to work on this issue.

**Comment by sagar10arya** [comment]
@tbaederr sir maybe this warning is showed because compilers does not support applying attributes to the underlying type of an enum class. Is it possible? Can you suggest something.

**Comment by sagar10arya** [comment]
> The attributes just need to go after 'enum class': https://godbolt.org/z/441Pnq64G

@tbaederr  Yaa, I was doing the same thing. It has been resolved.

**Comment by tbaederr** [comment]
I haven't seen another patch so yes.

**Comment by Krishna-13-cyber** [comment]
I was working on the patch as well but did not mention in this thread! You can continue @sagar10arya, I will revert back.
GCC rightly captures this warning at place,very useful for clang as well.
![Screenshot from 2023-03-29 14-29-33](https://user-images.githubusercontent.com/84722531/228484486-47258f27-4e10-470b-9e61-acb3a1a8e8ba.png)

**Comment by ipriyanshi1708** [comment]
I am an Outreachy applicant and I want to work on this issue.

**Comment by ipriyanshi1708** [comment]
Can anyone please tell me that whether TST_enum_class is a predefined declaration specifier or not? If not, then how can I identify enum class?

**Comment by krishenm94** [comment]
@sagar10arya hey, are you still working on this? If not I could try to pick it up

**Comment by ipriyanshi1708** [comment]
> @sagar10arya hey, are you still working on this? If not I could try to pick it up

Hi! @krishenm94, I have resolved the issue. Now I am working on the tests. https://reviews.llvm.org/D147989
